assets/src/settings-page/template-modes.js

```diff
@@ -0,0 +1,119 @@
+import { AMPNotice, NOTICE_SIZE_LARGE, NOTICE_TYPE_INFO, NOTICE_TYPE_WARNING } from '../components/amp-notice';
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Provides the notice to show in the reader theme support mode selection.
+ *
+ * @param {string} themeSupport The current theme support mode.
+ */
+function getReaderNotice( themeSupport ) {
+	switch ( true ) {
+		// Theme has built-in support or has declared theme support with the paired flag false.
+		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || ( 'object' === typeof THEME_SUPPORT_ARGS && false === THEME_SUPPORT_ARGS.paired ) ):
+			return (
+				<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_WARNING }>
+					<p>
+						{ __( 'Your active theme is known to work well in standard mode.', 'amp' ) }
+					</p>
+				</AMPNotice>
+			);
+
+		// Theme has built-in support or has declared theme support with the paired flag true.
+		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || ( 'object' === typeof THEME_SUPPORT_ARGS && true === THEME_SUPPORT_ARGS.paired ) ):
+			return (
+				<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_WARNING }>
+					<p>
+						{ __( 'Your active theme is known to work well in standard and transitional mode.', 'amp' ) }
+					</p>
+				</AMPNotice>
+			);
+
+		// Support for reader mode was detected.
+		case THEME_SUPPORTS_READER_MODE:
+			return (
+				<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_INFO }>
+					<p>
+						{ __( 'Your theme indicates it works best in reader mode.', 'amp' ) }
+					</p>
+				</AMPNotice>
+			);
+
+		default:
+			return null;
+	}
+}
+
+/**
+ * Template modes section of the settings page.
+ */
+export function TemplateModes() {
+	const { editedOptions } = useContext( Options );
+	const { plugin_configured: pluginConfigured, theme_support: themeSupport } = editedOptions;
+
+	return (
+		<section className="template-mode-selection">
+			<h2>
+				{ __( 'Template mode', 'amp' ) }
+			</h2>
+			<p dangerouslySetInnerHTML={ {
+				__html: __( 'For a list of themes and plugins that are known to be AMP compatible, please see the <a href="https://amp-wp.org/ecosystem/">ecosystem page</a>.', 'amp' ),
+			} } />
+			<TemplateModeOption
+				details={ __( 'In Standard Mode your site uses a single theme and there is a single version of your content. In this mode, AMP is the framework of your site and there is reduced development and maintenance costs by having a single site to maintain.', 'amp' ) }
+				mode="standard"
+			>
+				{
+					// Plugin is not configured; active theme has built-in support or has declared theme support without the paired flag.
+					! pluginConfigured && ( THEME_IS_SUPPORTED || 'object' === typeof THEME_SUPPORT_ARGS ) && (
+						<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_INFO }>
+							<p>
+								{ __( 'Your active theme is known to work well in standard mode.', 'amp' ) }
+							</p>
+						</AMPNotice>
```

Comment:
We should ensure that `paired` is not a property on `THEME_SUPPORT_ARGS`:

```suggestion
					! pluginConfigured &&
					(
						THEME_IS_SUPPORTED ||
						( 'object' === typeof THEME_SUPPORT_ARGS && ! THEME_SUPPORT_ARGS.hasOwnProperty( 'paired' ) )
					) &&
					(
```